Code/Framework/AzCore/AzCore/IO/Streamer/RequestPath.cpp

```diff
@@ -37,44 +37,28 @@ namespace AZ
 
         bool RequestPath::operator==(const RequestPath& rhs) const
         {
-            if (m_path.empty() || rhs.m_path.empty())
-            {
-                return m_path.empty() == rhs.m_path.empty();
-            }
-
-            ResolvePath();
-            rhs.ResolvePath();
-            if (m_absolutePathHash != rhs.m_absolutePathHash)
-            {
-                return false;
-            }
-            else
-            {
-                return m_path == rhs.m_path;
-            }
+            return m_absolutePathHash != rhs.m_absolutePathHash ? false : (m_path == rhs.m_path);
         }
 
         bool RequestPath::operator!=(const RequestPath& rhs) const
         {
-            return !operator==(rhs);
+            return m_absolutePathHash != rhs.m_absolutePathHash ? true : (m_path != rhs.m_path);
```

Comment:
Why was this call changed from calling `!operator==()`?

This seems like an easy candidate where bugs can be introduced because the comparison isn't symmetric.

Reply:
This was a left over of another experiment that didn't pan out. Should have been reverted, thanks for pointing it out.